aop/src/main/java/io/micronaut/aop/chain/InterceptorChain.java

```diff
@@ -121,36 +121,76 @@ public R proceed() throws RuntimeException {
      * @param beanContext bean context passed in
      * @param method The method
      * @param interceptors The array of interceptors
+     * @param <T> The intercepted type
      * @return The filtered array of interceptors
      */
     @SuppressWarnings("WeakerAccess")
     @Internal
     @UsedByGeneratedCode
-    public static Interceptor[] resolveAroundInterceptors(
-            @Nullable BeanContext beanContext,
-            ExecutableMethod<?, ?> method,
-            List<BeanRegistration<Interceptor<?, ?>>> interceptors) {
+    public static <T> Interceptor<T, ?>[] resolveAroundInterceptors(BeanContext beanContext,
```

Comment:
No signature changes to the old method, just some generics added to satisfy the compiler.